scalability_and_performance/understanding-node-observability-operator.adoc

```diff
@@ -0,0 +1,313 @@
+* You have access to the {product-title} web console.
+
+.Procedure
+
+. Log in to the OpenShift Container Platform web console.
+. In the Administrator's navigation panel, expand *Operators* → *OperatorHub*.
+. In the *All items* field, enter *Node Observability Operator* and select the Node Observability Operator tile.
+. Click *Install*.
+. On the *Install Operator* page:
+.. *Update channel*: Select *alpha*.
+.. *Installation mode*: Select *A specific namespace on the cluster*.
+.. *Installed Namespace*: Choose *node-observability-operator* from the list.
+.. *Update approval*: Select *Automatic*.
+.. Click *Install*.
+
+.Verification
+. In the Administrator's navigation panel, expand *Operators* → *Installed Operators*.
+. Verify that the Node Observability Operator is listed in the Operators list.
+
+== Creating the Node Observability custom resource
+Before you run profiling queries, you must create a `NodeObservability` custom resource (CR).
+
+When you apply the the `NodeObservability` CR, it creates the necessary machine config and machine config pool CRs to enable the CRI-O profiling on selected worker nodes.
+
+[NOTE]
+====
+Kubelet profiling is enabled by default.
+====
+
+.Prerequisites
+* The Node Observability Operator has been installed.
+* You have access to the cluster with `cluster-admin` privileges.
+
+.Procedure
+
+. Log in to the {product-title} CLI as a user with the `cluster-admin` role:
++
+[source,terminal]
+----
+$ oc login -u kubeadmin https://<HOSTNAME>:6443
+----
+
+. Create a CR file named `nodeobservability.yaml` that contains the following text:
++
+[source,yaml]
+----
+    apiVersion: nodeobservability.olm.openshift.io/v1alpha1
+    kind: NodeObservability
+    metadata:
+      name: cluster <1>
+    spec:
+      labels:
+        node-role.kubernetes.io/worker: ""
+      type: crio-kubelet
+----
+<1> You must specify the name as `cluster` because there should be only one `NodeObservability` CR per cluster.
+
+. Run the `NodeObservability` CR:
++
+[source,terminal]
+----
+oc apply -f nodeobservability.yaml
+----
+
++
+.Example output
+[source,terminal]
+----
+nodeobservability.olm.openshift.io/cluster created
+----
+
+. Review the status of the `NodeObservability` CR:
++
+[source,terminal]
+----
+$ oc get nob/cluster -o yaml | yq '.status'
+----
+
++
+.Example output
+[source,terminal]
+----
+conditions:
+  conditions:
+  - lastTransitionTime: "2022-07-05T07:33:54Z"
+    message: 'DaemonSet node-observability-ds ready: true NodeObservabilityMachineConfig
+      ready: true'
+    reason: Ready
+    status: "True"
+    type: Ready
+    count: 1
+    lastUpdated: "2022-07-05T07:33:55Z"
+----
+
++
+`NodeObservability` CR run is completed when the reason `Ready` and status `True`.
+
+
+== Running profiling queries
+
+Profiling query is a blocking operation that fetches CRI-O and Kubelet profiling for a duration of 30 seconds. To request a profiling query, you have to create a `NodeObservabilityRun` resource.
```

Comment:
> for a duration of 30 seconds
Cannot find this timeout anywhere. You meant the duration of the block which the query puts on or the duration of the profiling data?

Reply:
@alebedev87 I meant duration of the block which the query puts. If this is causing confusion, please suggest a rewrite.

Reply:
Ah, ok I see [it in the code](https://github.com/openshift/node-observability-agent/blob/main/pkg/handlers/handlers.go#L223). This is a timeout which gives up the query. It's set to 35 seconds actually. 
Well we can remove the mention of the seconds at all, or we can say that it may block up to `35` seconds max.

Reply:
For background, the 30s are probably coming from the duration of the pprof command. 
it's true that 30s is the time it takes when all goes right, and maxed at 35s in case of problem

Reply:
Yes, true, 30 seconds is [the default probing period for the pprof handler](https://pkg.go.dev/net/http/pprof#Profile) integrated into kubelet.
So, then if I read the phrase right - 30s is the duration of the probe, not the blocking period.

Reply:
Sorry, about the confusion. The phrase is fine as it was from the beginning.